src/components/menu-bar/project-title-input.jsx

```diff
@@ -0,0 +1,61 @@
+import classNames from 'classnames';
+import {connect} from 'react-redux';
+import PropTypes from 'prop-types';
+import bindAll from 'lodash.bindall';
+import React from 'react';
+
+import BufferedInputHOC from '../forms/buffered-input-hoc.jsx';
+import Input from '../forms/input.jsx';
+const BufferedInput = BufferedInputHOC(Input);
+
+import {setProjectTitle} from '../../reducers/project-title';
+
+import styles from './project-title-input.css';
+
+class ProjectTitleInput extends React.Component {
+    constructor (props) {
+        super(props);
+        bindAll(this, [
+            'handleUpdateProjectTitle'
+        ]);
+    }
+    // call onUpdateProjectTitle if it is defined (only defined when gui
+    // is used within scratch-www)
+    handleUpdateProjectTitle (newTitle) {
+        this.props.onSetProjectTitle(newTitle);
```

Comment:
I think it'd be more correct if this were something like:
```jsx
        if (this.props.onUpdateProjectTitle) {
            this.props.onUpdateProjectTitle(newTitle);
        } else {
            this.props.onSetProjectTitle(newTitle);
        }
```

That way the redux state won't get out of sync from the incoming prop, and changing the title would work without configuration.

When `onUpdateProjectTitle` is supplied, calling it should cause the parent component (www or the gui playground) to pass in a new title prop to the GUI component, and then that change should dispatch the redux state change, causing everything looking at the redux state to update.

Reply:
I also think there's an issue with the naming here. It's not clear what the difference is between `onUpdateProjectTitle` and `onUpdateProjectTitle`. I think it's because they're the same thing. If we move to the behavior I outlined above, then I think it would be possible to collapse these into a single prop.

Reply:
Done.